internal/dispatch/graph/computecheck_test.go

```diff
@@ -737,6 +748,47 @@ func TestComputeCheckWithCaveats(t *testing.T) {
 				},
 			},
 		},
+		{
+			"context type error test",
+			`definition user {}
+
+			definition document {
+				relation viewer: user
+
+				permission view = viewer
+			}`,
+			map[string]caveatDefinition{
+				"testcaveat": {
+					"somecondition == 42",
+					map[string]types.VariableType{
+						"somecondition": types.UIntType,
+					},
+				},
+			},
+			[]caveatedUpdate{
+				{core.RelationTupleUpdate_CREATE, "document:foo#viewer@user:sarah", "testcaveat", nil},
+			},
+			[]check{
+				{
+					"document:foo#view@user:sarah",
+					map[string]any{
+						"somecondition": "43a",
+					},
+					v1.ResourceCheckResult_NOT_MEMBER,
+					[]string{},
+					"type error for parameters for caveat `testcaveat`: could not convert context parameter `somecondition`: for uint: a uint64 value is required, but found invalid string value `43a`",
```

Comment:
I wonder if this is something we should consider scrubbing from the logs. We don't know what users are going to store in the context.

Reply:
its a tradeoff between data and UX

Reply:
It certainly is but for some engineering organizations this is not negotiable for compliance reasons. I guess we can _cross that bridge when we get there_.

Reply:
To be clear: this would only return data provided in the check context anyway back to the caller as part of an error message; it wouldn't leak it elsewhere unless the caller is logging it themselves or trace logging was on in the server (which would show a bunch of other call info anyway)